components/omega/doc/design/ShallowWaterOmega0.md

```diff
@@ -0,0 +1,212 @@
+
+### 2.3 Requirement: Omega-0 will use MPAS format unstructured-mesh domains
+
+We will continue to use the MPAS-format netcdf files for input and output, with the same mesh variable names and dimensions. This will facilitate ease of use and interoperability between MPAS-Ocean and Omega.
+
+Omega mesh information will be stored in a separate mesh file, and not be included with initial condition, restart, or output files. This way there is never redundant mesh data stored in these files.
+
+To do: add list of minimum number of variables that defines the mesh, and the additional variables that can be computed on start-up.
+
+### 2.4 Requirement: Omega-0 will interface with polaris for preprocessing and postprocessing
+
+A substantial investment has already been made in the polaris tools. Continuing the use of polaris for Omega will speed development and encourage the documentation and long-term reproducibility of test cases.
+
+The test cases relevant to this design document are in Section 5 below.
+
+### 2.5 Requirement: Omega-0 will run portably on various DOE architectures (CPU and GPU nodes)
+
+Omega will be able to run on all the upcoming DOE architectures and make good use of GPU hardware. This should occur with minimal alterations in the high-level PDE solver code for different platforms. 
+
+The proposed solution is to use YAKL [(Norman et al. 2023)](https://link.springer.com/10.1007/s10766-022-00739-0) for within-node, shared memory computations. All large arrays will be YAKL objects, and computations that loop over array indices will use YAKL function calls.
+
+### 2.6 Requirement: Omega-0 will run on multi-node with domain decomposition
+
+This is with MPI and halo communication, as described in framework design documents.
+
+### 2.7 Requirement: Correct convergence rates of operators and exact solutions
+
+Each operator will be tested individually for the proper convergence rate: divergence, gradient, curl interpolated to cell centers, and tangential velocity are all second order; curl at vertices is first order. The details of the convergence tests are explained in section 5.1 below.
+
+### 2.8 Requirement: Conservation of volume and tracer
+
+The total volume of the domain should be conserved to machine precision in the absence of surface volume fluxes. Likewise, the total tracer amount should be conserved to machine precision without surface fluxes. A simple test is to initialize an inactive test tracer with a uniform value of 1.0, and it should remain 1.0 throughout the domain.
+
+### 2.9 Requirement: Performance will be at least as good or better than MPAS-Ocean
+
+Performance will assessed with the following metrics:
+
+1. Single CPU throughput
+1. Parallel CPU scalability to high node counts
+1. Single GPU throughput
+
+The first two items will be compared to MPAS-Ocean with the same test configuration.
```

Comment:
Is it worth describing what we mean by MPAS-Ocean in the same configuration, E.g. what terms, parameterizations, etc. are on and what are turned off?

Reply:
Thanks. I added those details. Conveniently, this will be published soon in the Bishnu GMD paper.